### DISCUSSION TITLE
parallel inference for multiple requests

### DISCUSSION THREAD

**Discussion by ooaykac** [ggml-org/llama.cpp#10329] [Q&A]
What should I do to enable multiple users to ask questions to the language model simultaneously and receive responses? Does llama.cpp support parallel inference for concurrent operations?

How can we ensure that requests made to the language model are processed and inferred in parallel, rather than sequentially, to serve multiple users simultaneously?

**Comment by lukaLLM** [comment]
+1

**Comment by mtmcp** [comment]
[This discussion](https://github.com/ggerganov/llama.cpp/discussions/4130) on batch processing may provide some guidance.
